dig/dig_test.go

```diff
@@ -185,6 +185,21 @@ func TestConcurrentAccess(t *testing.T) {
 	}
 }
 
+func TestCycles(t *testing.T) {
```

Comment:
Can you add a test with one more level in between? Make sure it can resolve a loop with more elements in it.